lib/actions/createEvent.js

```diff
@@ -1,78 +1,39 @@
-
-function getCalendars(cfg, cb) {
-
-    function processData(items) {
-        console.log('Processing calendar data ...');
-        let result = {};
-        _.forEach(items.value, function setItem(item) {
-            result[item.id] = item.name;
-        });
-        return result;
-    }
-
-
-    console.log('Getting calendar data ...');
-    const instance = new ApiClient(cfg);
-    return instance
-        .get('/me/calendars')
-        .then(processData)
-        .nodeify(cb);
-}
-
-
-function processAction(msg, cfg) {
-    const self = this;
-    const calendarId = cfg.calendarId;
-    console.log(`About to create event in calendar with id=${calendarId}`);
-
-    const apiCall = `/me/calendars/${calendarId}/events`;
-
-    const instance = new ApiClient(cfg, self);
-
-    function createEvent(postRequestBody) {
-        console.log('Creating Event with properties:');
-        console.log(JSON.stringify(postRequestBody, null, 2));
-        return instance.post(apiCall, postRequestBody);
-    }
-
-    function emitData(data) {
-        const id = data.id;
-        console.log('Successfully created event with ID=' + id);
-        const messageBody = _.omitBy(data, (value, key) => key.startsWith('@odata.'));
-        messageBody.calendarId = cfg.calendarId;
-        console.log('Emitting data ...');
-        console.log('%j', messageBody);
-        self.emit('data', messages.newMessageWithBody(messageBody));
-    }
-
-    function emitError(e) {
-        console.log('Oops! Error occurred');
-        self.emit('error', e);
-    }
-
-    function emitEnd() {
-        console.log('Finished execution!');
-        self.emit('end');
-    }
-
-
-    let promise = processEventData(cfg, msg.body)
-        .then(createEvent)
-        .then(emitData)
-        .fail(emitError);
-
-    return promise.finally(emitEnd);
-
-}
-
-module.exports.process = processAction;
-module.exports.getCalendars = getCalendars;
-
+const { processEventData } = require('../processEventDataHelper');
+const { Client } = require('../Client');
+
+exports.process = async function process(msg, cfg) {
+  const client = new Client(this, cfg);
+  const { calendarId } = cfg;
+  this.logger.info(`About to create event in calendar with id=${calendarId}`);
+  const postRequestBody = await processEventData(cfg, msg.body);
+  this.logger.trace('Created postRequestBody: %j', postRequestBody);
+  try {
+    const data = await client.createEvent(calendarId, postRequestBody);
+    this.logger.info(`Successfully created event with ID=${data.id}`);
+    const messageBody = _.omitBy(data, (value, key) => key.startsWith('@odata.'));
+    messageBody.calendarId = cfg.calendarId;
+    this.logger.info('Emitting data ...');
+    this.logger.trace('Emitting data: %j', messageBody);
+    await this.emit('data', messages.newMessageWithBody(messageBody));
+  } catch (e) {
+    this.logger.error('Oops! Error occurred');
+    this.emit('error', e);
```

Comment:
I think, better to exchange the data inside these operations:
```js
    this.logger.error('Error occurred: ' + e);
    this.emit('error', 'Oops! Error occurred');
```

Reply:
In old version it was:
```
function emitError(e) {
        console.log('Oops! Error occurred');
        self.emit('error', e);
    }
```
let's leave it as is